.github/workflows/tpu-testing.yml

```diff
@@ -0,0 +1,96 @@
+name: Run TPU tests on current PR.
+
+on:
+  pull_request:
+    branches:
+    - master
+
+env:
+  PROJECT_ID: ${{ secrets.GKE_PROJECT }}
+  GKE_CLUSTER: oneshots-europe-west4   # TODO: update to cluster name.
+  GKE_ZONE: europe-west4-a  # TODO: update to cluster zone.
+  GKE_REGION: europe-west4  # TODO: update to cluster region.
+  IMAGE: gcr.io/${{ secrets.GKE_PROJECT }}/tpu-testing-image
+
+jobs:
+  setup-build-publish-deploy:
+    name: tpu-testing-job
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.14.x
+
+    - name: Checkout Pytorch Lightning
+      uses: actions/checkout@v2
+      with:
+        repository: PyTorchLightning/pytorch-lightning
+        ref: ${{ github.event.pull_request.head.sha }}
+        path: main
+
+    - name: Checkout ml-testing-accelerators
+      uses: actions/checkout@v2
+      with:
+        repository: GoogleCloudPlatform/ml-testing-accelerators
+        path: ml-testing-accelerators
+        ref: 4ebcf060a110a359c2f214d81be7334313ce3c03
+
+    # Setup gcloud CLI
+    - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+      with:
+        version: '290.0.1'
+        service_account_key: ${{ secrets.GKE_SA_KEY_BASE64 }}
+        project_id: ${{ secrets.GKE_PROJECT }}
+        export_default_credentials: true
+
+    # Configure Docker to use the gcloud command-line tool as a credential
+    # helper for authentication
+    - run: |-
+        gcloud --quiet auth configure-docker
+
+    - name: Build and Push Docker Image
+      run: |-
+        cd main/docker/tpu && \
+        docker build --tag "$IMAGE:$GITHUB_RUN_ID" -f Dockerfile --build-arg "GITHUB_REF=$GITHUB_REF" --build-arg "TEST_IMAGE=1" . && \
```

Comment:
thinking that the docker building on every CI would be quite slow, so shall we have a base (all steps before clone repo) and pull it from a docker hub? I guess that GH action does not have caches or existing layers

Reply:
When testing this locally, build+push was taking ~6m40s. I think we could have a base image that just installs the pytorch/xla portion and then during the test, we just pull latest `pytorch-lightning` code and install that.

We would need to set up some workflow to build that base image every day since pytorch/xla publishes new wheels every day. However in the near future, pytorch/xla should have a good stable version available so we could use the same base image repeatedly.

I'm not sure how much time this would save - to me it looks like the pulling and pushing of the image are the slow parts and I think we'd still need those 2 steps even if some of the steps were pushed to a base image.